forge-gui/res/cardsfolder/upcoming/banishment.txt

```diff
@@ -0,0 +1,8 @@
+Name:Banishment
+ManaCost:3 W
+Types:Enchantment
+K:Flash
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ When CARDNAME enters the battlefield, exile target nonland permanent an opponent controls and all other nonland permanents that player controls with the same name as that permanent until CARDNAME leaves the battlefield.
+SVar:TrigExile:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | ValidTgts$ Permanent.nonLand+OppCtrl | TgtPrompt$ Select target nonland permanent an opponent controls | SubAbility$ DBChangeZoneAll | Duration$ UntilHostLeavesPlay
+SVar:DBChangeZoneAll:DB$ ChangeZoneAll | Origin$ Battlefield | Destination$ Exile | ChangeType$ Permanent.nonLand+NotDefinedTargeted+sharesNameWith Targeted+ControlledBy TargetedOrController | Duration$ UntilHostLeavesPlay
```

Comment:
@Northmoc @tool4ever can you look into this? my Gut is telling me that the target and the one with the same name should be exiled at the same time

Reply:
What about:
 - eliminate line 6
 - rename line 7 `TrigExile`
 - add targeting to line 7
 - remove `NotDefinedTargeted`
 Seems like that would be cleaner?

Reply:
> What about:
> 
> * eliminate line 6
> * rename line 7 `TrigExile`
> * add targeting to line 7
> * remove `NotDefinedTargeted`
>   Seems like that would be cleaner?

I tried that but then Banishment only exiled one card and not all with the same name.
Note:I am a novice at scripting to be honest

Reply:
ChangeZoneAll probably can't target Cards
There are a few others with the same wording and while they don't have an explicit ruling it seems to imply simultaneously

Reply:
I think for now it might be good enough to just use Pump helper for targeting in Line 6 and try `ChangeType$ Targeted.Self,Permanent.nonLand+NotDefinedTargeted+sharesNameWith Targeted+ControlledBy TargetedController` in 7

Reply:
Please try like this:

```
SVar:TrigExile:DB$ Pump | ValidTgts$ Permanent.nonLand+OppCtrl | TgtPrompt$ Select target nonland permanent an opponent controls | SubAbility$ DBChangeZoneAll
SVar:DBChangeZoneAll:DB$ ChangeZoneAll | Origin$ Battlefield | Destination$ Exile | ChangeType$ TargetedCard.Self,Permanent.nonLand+NotDefinedTargeted+sharesNameWith Targeted+ControlledBy TargetedController | Duration$ UntilHostLeavesPlay
```